homeassistant/components/goalzero/__init__.py

```diff
@@ -0,0 +1,140 @@
+"""The Goal Zero Yeti integration."""
+import asyncio
+import logging
+
+from goalzero import Yeti, exceptions
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    DATA_KEY_API,
+    DATA_KEY_COORDINATOR,
+    DEFAULT_NAME,
+    DOMAIN,
+    MIN_TIME_BETWEEN_UPDATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+GOALZERO_SCHEMA = vol.Schema(
+    vol.All(
+        {
+            vol.Required(CONF_HOST): cv.matches_regex(
+                r"\A(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2 \
+            [0-4][0-9]|[01]?[0-9][0-9]?)\Z"
+            ),
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        },
+    )
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [GOALZERO_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+PLATFORMS = ["binary_sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the Goal Zero Yeti component."""
+
+    hass.data[DOMAIN] = {}
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Goal Zero Yeti from a config entry."""
+    name = entry.data[CONF_NAME]
+    host = entry.data[CONF_HOST]
+
+    _LOGGER.debug("Setting up %s integration with host %s", DOMAIN, host)
+
+    session = async_get_clientsession(hass)
+    api = Yeti(host, hass.loop, session)
+    try:
+        await api.get_state()
+    except exceptions.ConnectError as ex:
+        _LOGGER.warning("Failed to connect: %s", ex)
+        raise ConfigEntryNotReady from ex
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await api.get_state()
+        except exceptions.ConnectError as err:
+            _LOGGER.warning("Failed to update data from Yeti")
```

Comment:
There will be an error log by the coordinator when raising the `UpdateFailed` error. We don't need to log a warning too.